src/sql/src/plan/statement/dml.rs

```diff
@@ -886,6 +950,35 @@ pub fn plan_copy(
             }
             _ => sql_bail!("COPY FROM {} not supported", target),
         },
+        (CopyDirection::To, CopyTarget::Url(url_expr)) => {
+            scx.require_feature_flag(&vars::ENABLE_COPY_TO_URL)?;
+
+            let from = match relation {
+                CopyRelation::Table { name, columns } => {
+                    if !columns.is_empty() {
+                        // TODO(mouli): Add support for this
+                        sql_bail!("Specifying columns for COPY <table_name> TO commands not yet supported. Please use COPY (SELECT...) TO ... instead");
+                    }
+                    CopyToFrom::Id {
+                        id: *name.item_id(),
+                    }
+                }
+                CopyRelation::Select(stmt) => {
+                    if !stmt.query.order_by.is_empty() {
```

Comment:
For ORDER BY, we decided not to include it because it would be expensive https://github.com/MaterializeInc/materialize/pull/23556#discussion_r1419943983.

Good point about LIMIT. I think it could be supported. @benesch Does that sound right to you? Will update the design doc then.